include/zephyr/xen/dom0/domctl.h

```diff
@@ -12,27 +19,221 @@
 
 #include <zephyr/kernel.h>
 
+/**
+ * @brief Perform a scheduler operation on a specified domain.
+ *
+ * @param domid The ID of the domain on which the scheduler operation is to be performed.
+ * @param[in,out] sched_op A pointer to a `struct xen_domctl_scheduler_op` object that defines
+ *                         the specific scheduler operation to be performed.
+ * @return Returns 0 on success, or a negative error code on failure.
```

Comment:
Please reflect the suggestions from the review on #95525.
(Abut @return and @retval.)